src/NanoMsg.cpp

```diff
@@ -0,0 +1,49 @@
+#include "NanoMsg.h"
+#include <cstring>
+#include <stdexcept>
+#include <g3log/g3log.hpp>
+/**
+ * Constructor for copying a packet hash vector into a nanomsg buffer
+ */
+NanoMsg::NanoMsg(PacketHashVector& data) : sent(false) {
+   auto data_size = data.size() * sizeof(std::pair<void*, unsigned int>);
+   buffer = nn_allocmsg(data_size, 0);
+   auto error = nn_errno ();
+   if (error == ENOMEM) {
+      throw std::runtime_error("not enough memory to allocate NanoMsg: " +
+                               std::string(nn_strerror(errno)));
+   }
+   std::memcpy(buffer, &data[0], data_size);
+}
+NanoMsg::NanoMsg(void*& data) : sent(false) {
+   auto data_size = sizeof(void*);
+   buffer = nn_allocmsg(data_size, 0);
+   auto error = nn_errno ();
```

Comment:
this is fine. you could have used ```std::errno``` also. the ```nn_errno``` is only needed on Windows, ref: http://nanomsg.org/v0.2/nn_errno.3.html